inference-engine/src/transformations/src/transformations/op_conversions/convert_slice_to_strided_slice.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "transformations/op_conversions/convert_slice_to_strided_slice.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/validation_util.hpp"
+
+#include "itt.hpp"
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SliceToStridedSlice, "SliceToStridedSlice", 0);
+
+namespace {
+    Output<ngraph::Node> align_indices(const Output<ngraph::Node>& indices,
+                                   const Output<ngraph::Node>& slice_axes,
+                                   const Output<ngraph::Node>& scatter_axis,
+                                   size_t slice_indices_length,
+                                   int64_t fill_in_value,
+                                   NodeVector& new_ops) {
+    // Handle a case when starts/ends/steps lengths are less than provided axes
+    // in order to ensure compatibility with `StridedSlice:v1` interface
+    // Example:
+    // data_shape: {3, 3, 3, 3}
+    // starts: [1, 1] - after extending --> [0, 0, 1, 1]
+    // ends: [2, 2] - after extending --> [0, 0, 2, 2]
+    // steps : [1, 1] - after extending --> [1, 1, 1, 1]
+    // axes: [2, 3] - apply slice values to 2 and 3 dimension of input data
+    // expected_output_shape: {3, 3, 1, 1}
+
+    const auto default_indices = ngraph::opset8::Constant::create(indices.get_element_type(), Shape{slice_indices_length}, {fill_in_value});
+    std::shared_ptr<ngraph::Node> adjusted_indices = std::make_shared<ngraph::opset8::ScatterUpdate>(default_indices,
+                                                                                         slice_axes,
+                                                                                         indices, // updates
+                                                                                         scatter_axis);
+    std::shared_ptr<ngraph::Node> const_indices = get_constant_from_source(adjusted_indices);
```

Comment:
As you already checked inputs for the Constants [here](https://github.com/openvinotoolkit/openvino/pull/8295/files?authenticity_token=FyKtJagL%2BC7E4Rt7RbkitOokrq4xWSH3Is5Sse1TUisvE5iMQLABkpxCIc1eskBmcJKpNgCFMNdS6FUjQKGnTQ%3D%3D&file-filters%5B%5D=.cpp#diff-2fa79f60eb9ae62def921c5c97d142e46cabdbec4c45044712d84a825094be97R82) so you can simply use make_try_fold here and don't care about `m_use_shapes` key because `indices` will be either constant so method will return you Constant or ScatterUpdate otherwise.

Reply:
Changed to `make_try_fold`.